### ISSUE TITLE
glfwInit(), segfaults on AMD graphic card on linux

### ISSUE THREAD

**Issue by ikamthania** [glfw/glfw#835]
Using amd drivers amdgpu-pro_16.30.3 and on ubuntu 16.04 LTS. Everything in the tests folder fails with the message Segmentation fault (core dumped).

**Comment by ikamthania** [comment]
Something as simple as 
```
#define GLFW_INCLUDE_VULKAN
#include <GLFW/glfw3.h>
int main() {
    glfwInit();
    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
    GLFWwindow* window = glfwCreateWindow(800, 600, "window", nullptr, nullptr);
    glfwDestroyWindow(window);
    glfwTerminate();
    return 0;
}
```
returns with Segmentation fault (core dumped)

**Comment by ikamthania** [comment]
A similar issue was reported for Nvidia [here](https://github.com/glfw/glfw/issues/734). But I don't understand the solution there.

**Comment by joanasleet** [comment]
Glad I'm not the only one!

I can provide some additional infos.

Coredump:

```
Stack trace of thread 13367:
  #0  0x00007f00046abd22 amdgpu_query_gpu_info (libdrm_amdgpu.so.1)
  #1  0x00007f0006ccdb30 n/a (amdgpu_dri.so)
  #2  0x00007f0006cce2c4 n/a (amdgpu_dri.so)
  #3  0x00007f0006cd11d6 n/a (amdgpu_dri.so)
  #4  0x00007f000616b918 n/a (amdgpu_dri.so)
  #5  0x00007f0006116e44 n/a (amdgpu_dri.so)
  #6  0x00007f0006117047 n/a (amdgpu_dri.so)
  #7  0x00007f0004db1c9a n/a (amdgpu_dri.so)
  #8  0x00007f0005cbd717 n/a (amdgpu_dri.so)
  #9  0x00007f0006ccc1c0 n/a (amdgpu_dri.so)
  #10 0x00007f0006c58c78 n/a (amdgpu_dri.so)
  #11 0x00007f0006c58fd6 n/a (amdgpu_dri.so)
  #12 0x00007f0006c5399e n/a (amdgpu_dri.so)
  #13 0x00007f0006c5118f eglInitialize (amdgpu_dri.so)
  #14 0x00007f000b523a07 n/a (libglfw.so.3)
  #15 0x00007f000b51bf3f n/a (libglfw.so.3)
  #16 0x00007f000b5161ad glfwInit (libglfw.so.3)
  #17 0x0000000000428210 main (main)
  #18 0x00007f000a5c0291 __libc_start_main (libc.so.6)
  #19 0x000000000040336a _start (main)
```

System hangs on glfwInit().
- ArchLinux x64
- lates amdgpu-pro

Same with GLFW_INCLUDE_VULKAN.

glfwInit() and window creation succeed on glfw-3.1.2.

Other programs that use opengl work fine ( tested with Unigine Benchmark )

The gpu driver is tagged as beta, so it could be it.

**Comment by elmindreda** [comment]
This is very likely already fixed in `master`.

**Comment by joanasleet** [comment]
@elmindreda 
I just tested it - glfwInit() and window creation works in master!
Can you elaborate on the issue ? I'm curious.

Thanks for the quick solution.

**Comment by elmindreda** [comment]
The problem was (partially) resolved by d5e00e6b0fe1e64b4bb3b2cbe41018dad14f4e1f. The triggering factor is that on many systems was a bad idea to load both `libGL` and `libEGL`. I haven't had time to dig through the exact details yet.
